docs/github_projects_guidelines.md

```diff
@@ -44,68 +44,59 @@ working with projects hosted on GitHub.
-  Commits to it can be done to it directly or in form of GitHub Pull requests.
+Each repository has a single long-lived, default, branch named `main`.
 
-Besides these two branches, various short-lived branches exist in this workflow.
-They are classified into three categories: feature, release, and hotfix.
+There are two rules that should be followed when adding new code to the `main`:
 
-- feature branches are used to develop new features. They always branch from and
-  back into the `dev` branch, never in `master`. They are named as `feature/*`,
-  for example, the names `feature/new-board` and `feature/refactor` are both
-  valid feature branch names.
-- release branches are used for the preparation of a new release. They allow for
-  last-minute changes. They are always created from the `dev` branch and merged
-  into the `master` branch. They are named as `release/<version>`, where
-  `<version>` indicates the version of the next release.
-- hotfix branches are used to address unplanned changes to the release in
-  production. The essence is that the work of team members can continue, while
-  another person is preparing a quick production fix. Hotfix branches must
-  always branch from the `master` branch and merge back into it.
+- It should **never break the build** (whatever _build_ means for a specific
+  repository). Bugs should be caught by the CI or by code reviews.
+- `main` should **always be release ready**, even if there is work in progress.
+  Functionalities that might need some more development time to become useful
+  should not be included in the build until they are ready.
 
-It is important that after every merge into the `master` branch a Pull Request
-is made from `master` back into `dev`. This brings in the changes that were made
-during the release process in the release and hotfix branches.
+Those two rules can sometimes be hard to follow in practice due to many internal
+and external factors. If a deviation from those two rules happens effort should
+be made to bring the `main` branch into the correct state.
 
-### Deviations from GitFlow article
+New code can be added to the `main` in two ways:
 
-- Use `dev` instead of `develop`.
-- When tagging tags add `v-` prefix, so `v1.0.0` instead of `1.0.0`.
-- Use slash `/` instead of dash `-` when naming release, hotfix and feature
-  branches, so `release/v1.0.0` instead of `release-v1.0.0`.
+- either by committing to it directly, or
+- in the form of feature branches.
 
-### Feature branches
+#### Feature branches
 
 A few things should be kept in mind when working with feature branches:
 
-- They should be focused on accomplishing one thing.
-- They should be small, they should take up to 1 day of work to complete and to
-  open Pull Request back into the `dev` branch.
+- They should be focused on accomplishing one small thing.
+- They should take up to 1 day of work to complete and to open Pull Request back
+  into the `main` branch for the code review.
```

Comment:
```suggestion
- They should take up to 1 day of work to complete.
- Once complete, a Pull Request into `main` should be created for the code review.
```

About the "1 day of work" thing. Maybe add some text here explaining that larger work must be split into smaller tasks, that are planned to take up to 1 day of work.
Also, there are some that will take more than 1 day, and can not further be split (like bug-hunts for example) - how do we handle those?